sdk/keyvault/azure-security-keyvault-administration/src/main/java/com/azure/security/keyvault/administration/models/KeyVaultRoleAssignmentProperties.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.security.keyvault.administration.models;
+
+import com.azure.security.keyvault.administration.implementation.KeyVaultErrorCodeStrings;
+
+import java.util.Objects;
+
+/**
+ * A class that defines a role assignment's properties.
+ */
+public final class KeyVaultRoleAssignmentProperties {
+    private String roleDefinitionId;
+    private String principalId;
+
+    /**
+     * Creates a new {@link KeyVaultRoleAssignmentProperties role assignment properties} object with the specified
+     * details.
+     *
+     * @param roleDefinitionId The {@link KeyVaultRoleDefinition role definition} ID used in the
+     *                         {@link KeyVaultRoleAssignment role assignment}.
+     * @param principalId      The principal ID assigned to the role. This maps to the ID inside the Active Directory.
+     *                         It can point to a user, service principal, or security group.
+     */
+    public KeyVaultRoleAssignmentProperties(String roleDefinitionId, String principalId) {
```

Comment:
These are required parameters for create methods. Are they being passed to those methods directly as well?

Reply:
These parameters are passed directly to the service methods.

This Properties object is based on how the service Swagger groups parameters, however, I believe we certainly could change the method signature to take all required values like this:

```java
public Mono<KeyVaultRoleAssignment> createRoleAssignment(
    KeyVaultRoleAssignmentScope roleScope,
    String roleDefinitionId,
    String principalId)
```

Instead of what we have today:

```java
public Mono<KeyVaultRoleAssignment> createRoleAssignment(
    KeyVaultRoleAssignmentScope scope,
    KeyVaultRoleAssignmentProperties properties)
```

Reply:
I thought we were doing it that way in .NET after a discussion, but I don't see that in source. @christothes and @sadasant, was it a different method we were talking about pulling `roleDefinitionId` and `principalId` out of, or was that for the model constructor? I may just be remembering wrong here.

Reply:
I think we still use a properties bag in .NET, but I like the individual parameter shape better.